petastorm/tests/test_codec_scalar.py

```diff
@@ -76,3 +77,40 @@ def test_bad_unischema_field_shape():
     field = UnischemaField(name='field_int', numpy_dtype=np.int32, shape=(1,), codec=codec, nullable=False)
     with pytest.raises(ValueError, match='must be an empty tuple'):
         codec.encode(field, np.int32(1))
+
+
+def test_encode_scalar_bool():
+    codec = ScalarCodec(BooleanType())
+    field = UnischemaField(name='field_bool', numpy_dtype=np.bool, shape=(), codec=codec, nullable=False)
+
+    encoded = codec.encode(field, np.bool_(True))
+    assert isinstance(codec.encode(field, encoded), bool)
+    assert encoded
+
+    encoded = codec.encode(field, np.bool_(False))
+    assert not encoded
+
+
+def test_encode_scalar_int():
+    codec = ScalarCodec(IntegerType())
+    field = UnischemaField(name='field_int', numpy_dtype=np.int32, shape=(), codec=codec, nullable=False)
+    encoded = codec.encode(field, np.int32(42))
+    assert isinstance(encoded, int)
+    assert 42 == encoded
+
+
+def test_encode_scalar_float():
+    codec = ScalarCodec(FloatType())
+    expected = np.random.random(()).astype(np.float64)
+    field = UnischemaField(name='field_float', numpy_dtype=np.int32, shape=(), codec=codec, nullable=False)
+    encoded = codec.encode(field, expected)
+    assert isinstance(encoded, float)
+    assert expected == encoded
+
+def test_encode_scalar_string():
+    codec = ScalarCodec(StringType())
+    expected = 'surprise'
+    field = UnischemaField(name='field_string', numpy_dtype=np.unicode_, shape=(), codec=codec, nullable=False)
+    encoded = codec.encode(field, expected)
+    assert isinstance(encoded, str)
+    assert expected == encoded
```

Comment:
I am confused by this equivalence? Is this a value comparison, object_id comparison? 

What is the point of encoding if there is no change? Is this equivalence only true for primitive types?

Reply:
PySpark understands only basic Python types when you try to write scalar fields into a parquet store (e.g. int, float). 

In order to have rich scalar type set (e.g. float32, float64, int8, etc...), we present numpy scalars to user. Codec will take these numpy types and convert them to Python types understandable by Spark.

Reply:
okay understood